polaris-service/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1254,6 +1255,9 @@ public void doRefresh() {
     public void doCommit(TableMetadata base, TableMetadata metadata) {
       LOGGER.debug(
           "doCommit for table {} with base {}, metadata {}", tableIdentifier, base, metadata);
+      LOGGER.info(
+          "doCommit full new metadata: {}",
+          PolarisObjectMapperUtil.serialize(getCurrentPolarisContext(), metadata));
```

Comment:
Yea I'm both worried about the performance hit of serializing such a large object and the amount of data logged. Some possible options:
- Move to DEBUG (still performs serialization, does nothing for people who run in INFO+ which should be common)
- Gate the entire thing behind a featureConfiguration
- Set a limit on the log output length in featureConfiguration (still performs serialization)

What are your thoughts? I'm leaning toward the 2nd one.

Reply:
> Move to DEBUG (still performs serialization, does nothing for people who run in INFO+ which should be common)

Good callout. Man do I miss scala.

In this case, perhaps we can do something like:

```java
      LOGGER.info(
          "doCommit full new metadata: {}",
          () -> PolarisObjectMapperUtil.serialize(getCurrentPolarisContext(), metadata));
```

SLF4J 2.0 [supports](https://slf4j.org/manual.html#fluent) this, and in general it might be a useful wrapper for us.

Having said all of that, I'm less worried about the serde overhead, since we do so much metadata/property serde anyway, and more about emitting a huge blob into the logs. Particularly the INFO logs which should stay high-signal.

Reply:
Oh that's nice! I updated it to DEBUG and made it use the supplier syntax.